render-test/runner.cpp

```diff
@@ -294,6 +295,31 @@ bool TestRunner::checkRenderTestResults(mbgl::PremultipliedImage&& actualImage,
         }
     }
 
+    // Check network metrics.
+    for (const auto& expected : metadata.expectedMetrics.network) {
+        auto actual = metadata.metrics.network.find(expected.first);
+        if (actual == metadata.metrics.network.end()) {
+            metadata.errorMessage = "Failed to find network probe: " + expected.first;
+            return false;
+        }
+        if (actual->second.requests > expected.second.requests) {
+            std::stringstream ss;
+            ss << "Number of requests at probe \"" << expected.first << "\" is " << actual->second.requests
+               << ", expected is " << expected.second.requests << ".";
+
+            metadata.errorMessage = ss.str();
+            return false;
+        }
+        if (actual->second.transferred > expected.second.transferred) {
```

Comment:
Ditto. We will probably need to add tolerance if we make this `!=`.

Reply:
Done